src/token.rs

```diff
@@ -1,18 +1,66 @@
-use crate::location::Span;
+use std::fmt::Display;
+
+use crate::position::Span;
+
+#[derive(Debug)]
+pub enum RegSize {
+    Arch,  // architecure dependent
+    ByteLow,
+    ByteHigh,
+    Word,
+    DWord,
+    QWord,
+}
+
+impl From<Option<char>> for RegSize {
+    fn from(value: Option<char>) -> Self {
+        match value {
+            Some('l') => Self::ByteLow,
+            Some('h') => Self::ByteHigh,
+            Some('w') => Self::Word,
+            Some('d') => Self::DWord,
+            Some('q') => Self::QWord,
+            None => Self::Arch
+        }
+    }
+}
+
+impl Display for RegSize {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", match self {
+            Self::ByteLow  => "l",
+            Self::ByteHigh => "h",
+            Self::Word     => "w",
+            Self::DWord    => "d",
+            Self::QWord    => "q",
+            Self::Arch     => ""
+        })
+    }
+}
+
+#[derive(Debug)]
+pub enum IntLiteralRadix {
+    Bin,
+    Dec,
+    Hex
+}
 
 #[allow(clippy::upper_case_acronyms)]
 #[derive(Debug)]
-pub enum TokenKind {
+pub enum TokenKind<'a> {
+    Register(u8, RegSize),
+    Intager(&'a str, IntLiteralRadix),
+    StringLiteral(&'a str),
+    Identifier(&'a str),
+    CharLiteral(&'a str),
+
     Ampersand,
     At,
```

Comment:
maybe call this AtSign to avoid confusion with position?

Reply:
This is a good suggestion actually. Tokens are being almost completely reworked so ill make sure to rename it in my next commit